crates/core/src/abi.rs

```diff
@@ -307,6 +307,28 @@ def_instruction! {
             ty: TypeId,
         } : [2] => [1],
 
+        /// Pops all fields for a fixed list off the stack and then composes them
+        /// into an array.
+        FixedSizeListLift {
+            element: &'a Type,
+            size: u32,
+            id: TypeId,
+        } : [*size as usize] => [1],
+
+        /// Pops an array off the stack, decomposes the elements and then pushes them onto the stack.
+        FixedSizeListLower {
+            element: &'a Type,
+            size: u32,
+            id: TypeId,
+        } : [1] => [*size as usize],
```

Comment:
For this I'll echo some of my thoughts on the sibling PRs: I think we'll want to have this with a different shape where the size of the list isn't affecting the O(...) processing here. For example this is pushing/popping the list items from the operand stack of the ABI code, but I think we're going to want something different more akin to how dynamically sized lists work to avoid generating an exponential amount of code in the code generators

Reply:
Now, lifting from and lowering to memory uses constant code size, only when the list gets flattened in arguments the code remains O(N).

I will now fix the wasmtime implementation as well.